analyses/molecular-subtyping-LGAT/01-subset-files-for-LGAT.Rmd

```diff
@@ -0,0 +1,279 @@
+```{r}
+consensusMutationSubset %>% 
+  group_by(Hugo_Symbol,Variant_Classification) %>%
+  tally() 
+
+```
+ 
+We will remove synonymous variant calls from the list above. The Variant_Classification terms for synonymous SNV are selected as per [interaction-plots/scripts/02-process_mutations.R](https://github.com/AlexsLemonade/OpenPBTA-analysis/blob/master/analyses/interaction-plots/scripts/02-process_mutations.R).
+```{r}
+# Variant Classification with Low/Modifier variant consequences 
+#  from maftools http://asia.ensembl.org/Help/Glossary?id=535
+synonymous <- c(
+  "Silent",
+  "Start_Codon_Ins",
+  "Start_Codon_SNP",
+  "Stop_Codon_Del",
+  "De_novo_Start_InFrame",
+  "De_novo_Start_OutOfFrame"
+)
+
+consensusMutationSubset %>%
+  dplyr::filter(!Variant_Classification %in% synonymous)
+```
+
+Let's check whether all LGAT DNA biospecimens have a subtype defining SNV.
+ 
+```{r}
+all(lgat_dna_df$Kids_First_Biospecimen_ID %in% consensusMutationSubset$Tumor_Sample_Barcode)
+```
+ 
+### Gather status of all SNVs per biospecimen
+To gather the biospecimens, we will use all LGAT DNA biospecimen IDs and assign "Yes" or "No" to each subtype column depending on the presence or absence of SNV that define the subtype.
+
+columnname  | description 
+ --- | --- 
+NF1_mut | somatic loss of NF1 via either missense, nonsense mutation 
+BRAF_V600E_mut | contains BRAF V600E or V599 SNV or non-canonical BRAF alterations such as p.V600ins or p.D594N 
+MAPK_mut | contains mutation in KRAS, NRAS, HRAS, MAP2K1, MAP2K2, MAP2K1, ARAF SNV or indel 
+RTK_mut | harbors a MET, KIT, or PDGFRA SNV 
+FGFR_mut | harbors FGFR1 p.N546K, p.K656E, p.N577, or p. K687 hotspot mutations 
+IDH_mut | harbors an IDH R132 mutation 
+H3.1_mut | harbors an HIST1H3B K28M or HIST1H3C  K28M
+H3.2_mut | harbors an HIST2H3C K28M
+H3.3_mut | harbors an H3F3A K28M or G35R/V mutation 
+
+```{r}
+consensusMutationSubset <- lgat_dna_df %>%
+  left_join(consensusMutationSubset,by=c("Kids_First_Biospecimen_ID"="Tumor_Sample_Barcode")) %>%
+  reshape2::dcast(Kids_First_Biospecimen_ID + sample_id~ Hugo_Symbol) %>%
+  dplyr::mutate(BRAF_V600E_mut = if_else(BRAF>=1,"Yes","No"),
+                FGFR_mut = if_else(rowSums(
+                  dplyr::select(.,snvOI$FGFR$gene)) >=1,"Yes","No"),
+                # No IDH HIST1H3C, HIST2H3C, or H3F3A mutations found
+                IDH_mut  = if_else(any(colnames(.) %in% snvOI$IDH$gene) & 
+                                     rowSums(dplyr::select(.,one_of(snvOI$IDH$gene))) >=1,
+                                   "Yes","No"),
+                H3.1_mut = if_else(rowSums(
+                  dplyr::select(.,one_of(c("HIST1H3B","HIST1H3C")))) >=1 ,"Yes","No"), 
+                H3.2_mut = if_else(any(colnames(.) %in% snvOI$HIST1H3C$gene) & 
+                                     rowSums(dplyr::select(.,one_of(snvOI$HIST1H3C$gene))) >=1,
+                                   "Yes","No"), 
```

Comment:
should these rows have "snvOI$HIST2H3C" instead?

Reply:
oh yes, good catch, thanks!